src/core/symbology-ng/qgsfillsymbollayerv2.cpp

```diff
@@ -1571,6 +1571,16 @@ void QgsImageFillSymbolLayer::renderPolygon( const QPolygonF& points, QList<QPol
   applyDataDefinedSettings( context );
 
   p->setPen( QPen( Qt::NoPen ) );
+
+  if ( context.renderContext().testFlag( QgsRenderContext::RenderMapTile ) )
+  {
+    //transform brush to upper left corner of geometry bbox
+    QPointF leftCorner = points.boundingRect().topLeft();
+    QTransform leftCornerTransform = QTransform::fromTranslate( leftCorner.x(), leftCorner.y() );
+    mBrush.setTransform( leftCornerTransform );
```

Comment:
Looks like in some places a transform is already set on the brush. Might be safer to take the existing brush transform and then translate it (similar to how rotation is done in line 1593)